lib/mailgun/client.rb

```diff
@@ -12,13 +12,36 @@ class Client
     def initialize(api_key = Mailgun.api_key,
                    api_host = 'api.mailgun.net',
                    api_version = 'v3',
-                   ssl = true)
+                   ssl = true,
+                   test_mode = false)
```

Comment:
@kpandya91 So actually that's a really good point. It looks like `Mailgun.test_mode` was defined in an earlier commit but was never actually used? So this actually could make use of that field. I'll dig through the code again to make sure it's unused and add that as the default value. Thanks!